### PR TITLE
DOC: add About Us page

### PR DESCRIPTION
**Pull request by FFY00**
Fixes GH-182

### CODE REVIEW

**Review by rgommers — COMMENTED** [review]
Thanks @FFY00. Two comments
1. I think we should list the maintainers in a section above the institutional partners (credit where credit is due, so folks who do the heavy lifting first)
2. The logo you used is outdated, can you please find the new logo?

**Comment by FFY00** [comment]
> I think we should list the maintainers in a section above the institutional partners (credit where credit is due, so folks who do the heavy lifting first)

Yes, I was on the edge about it, but didn't include it because we already have the contributors section in the index, but I'll add it :+1: 

> The logo you used is outdated, can you please find the new logo?

I copied from NumPy as the ones from the website are not SVG, and I was only able to find up-to-date vectorized logos for Labs in our internal docs. Should I put Quansight Labs instead? Otherwise, I can ping someone to find an up-to-date SVG of the Quansight logo.

**Comment by rgommers** [comment]
I sent you a logo file. 

Also, in terms of giving credit, I think it's worth mentioning the Meson maintainers for all their help, and in particular Eli Schwartz, Dylan Baker and Jussi Pakkanen. I checked the Meson docs, but there's no list - all I could find in that vein is the 1.0.0 release blog post (https://nibblestew.blogspot.com/2022/12/after-exactly-10-years-meson-100-is-out.html)

**Comment by rgommers** [comment]
> because we already have the contributors section in the index

inserting a link to https://meson-python.readthedocs.io/en/latest/#contributors is also helpful.

File: docs/about-us.rst
Comment (rgommers) [inline docs/about-us.rst:13]: please remove (Expat), it's known as the MIT license

File: docs/about-us.rst
Comment (rgommers) [inline docs/about-us.rst:12]: Please remove the link from Python, and every repeated link to `Meson_` further down. The overuse of links, especially when they are repeatedly pointing to the exact same URL, hinders readability.

**Review by rgommers — COMMENTED** [review]
Thanks @FFY00, overall this looks quite good to me now, just a couple of minor points for readability. The "don't over-use links" one applies generally to writing docs I'd say.

**Comment by rgommers** [comment]
I plan to merge this soon, just a ping for @dnicolodi, @henryiii and @lithomas1 for visibility, since they're mentioned by name.

File: docs/about-us.rst
Comment (dnicolodi) [inline docs/about-us.rst:14]: Mentioning the Meson license here is probably important as it is obviously as required dependency and its license AFAIK is not compatible with the MIT license https://mesonbuild.com/legal.html

**Review by dnicolodi — CHANGES_REQUESTED** [review]
I agree with the spirit. I think the implementation could be better. This page would be better titled "Licensing and governance" or something like that, with sections "License", "Contributors" (or "Maintainers"), "Funding" (or something, but not "Institutional Partners").

**Comment by dnicolodi** [comment]
Why isn't the REUSE thing not complaining that the Quansight logo file hasn't an associate license or copyright statement?

File: docs/about-us.rst
Comment (FFY00) [reply docs/about-us.rst:18]: It is the last thing on the page, I don't really see an issue with it. I find that it gives a better sense of community to the project, and that was the reason I added it in the first place. This is purely subjective, and it seems we simply disagree, so I don't think it's very productive to argue about it. I am -1 on the proposal, and defer to the other maintainers for the final call.

File: docs/about-us.rst
Comment (FFY00) [reply docs/about-us.rst:30]: I disagree, some people know me from my name, some know me from my nick, so I think it's relevant. Regarding the list type, I think it looks nicer like this, which is why I wrote it as such. Similarly to above, this is subjective, and I am -1 and -0.5 on the proposals, respectively, so I defer to the other maintainers for the final call.

File: docs/about-us.rst
Comment (FFY00) [reply docs/about-us.rst:21]: Agreed, I'll make the change, thanks.

File: docs/about-us.rst
Comment (FFY00) [reply docs/about-us.rst:13]: I'm indifferent, so I'll make the change unless someone else also has an opinion on this.

File: docs/about-us.rst
Comment (FFY00) [reply docs/about-us.rst:12]: > "making it possible to it for Python_ packaging": something went wrong with this sentence.

Yes, thank you for noticing. I'll fix it.

> Also, this repeats what's in the index page. I think that anyone looking at this page would have an idea of what meson-python is, or they would not care either way.

This is supposed to be an expanded description of the project, which is relevant for people other than just the ones using it. For eg. I have received inquiries from people overseeing government software projects regarding my open source projects being used as a dependency, and having a page like this can either avoid them contacting us, or provide us a good page to point them to. This is just an example though, there are many other people that would be reading this page.

I don't really don't see a significant benefit removing the section, only the opposite, so I'd rather keep this section if this is not something you feel strongly about.

File: docs/about-us.rst
Comment (FFY00) [reply docs/about-us.rst:53]: https://meson-python--354.org.readthedocs.build/en/354/about-us.html

It looks okay, and IMO looks significantly better without it. I don't think it is productive to argue about this at all.

File: docs/about-us.rst
Comment (FFY00) [reply docs/about-us.rst:33]: Sure. I am indifferent, so I'll make the change unless someone else has an opinion on this.

File: pyproject.toml
Comment (FFY00) [reply pyproject.toml:61]: The `:user:``` role.

File: docs/about-us.rst
Comment (FFY00) [reply docs/about-us.rst:36]: This was pretty much copied from NumPy's documentation. I really don't want to argue about it, so I'll defer to @rgommers.

File: docs/about-us.rst
Comment (FFY00) [reply docs/about-us.rst:14]: Good, idea, thanks! Will make the change.

File: docs/about-us.rst
Comment (FFY00) [reply docs/about-us.rst:13]: Just stating MIT is ambiguous, we are using the Expat variant of the MIT license, and that's why I added it. I am pretty much indifferent, so I'll make the change unless someone else has an opinion.

File: docs/about-us.rst
Comment (dnicolodi) [reply docs/about-us.rst:12]: I don't want the section removed. Just the repetition of what meson-python is. This is not an "extended description", it is an (half) sentence that is not enough to understand what the project does to who does not know it already, and is useless repetition for all others.

File: docs/about-us.rst
Comment (dnicolodi) [reply docs/about-us.rst:36]: I think you refer to the "Institutional Partners" section on this document https://numpy.org/about/. This needs to be read paired with this https://numpy.org/doc/stable/dev/governance/governance.html#institutional-partners-and-funding Having to define the "Instituttional Partner" role to state that two of the maintainers of meson-python do that as part of their job seems useless indirection to me. I like to keep things simple.

File: docs/about-us.rst
Comment (dnicolodi) [reply docs/about-us.rst:13]: What? Our SPDX headers state that the license is MIT, not "MIT (Expat)". There is no "MIT (Expat)" license in the SPDX list of licenses https://spdx.org/licenses/ or in the OSI approved licenses https://opensource.org/licenses/

File: docs/about-us.rst
Comment (dnicolodi) [reply docs/about-us.rst:53]: So you agree it looks better without but you leave it there? I'm confused.

File: docs/about-us.rst
Comment (rgommers) [reply docs/about-us.rst:36]: Yes, the "Institutional Partners" thing is a specific term that is defined in the governance docs of not only NumPy, but Jupyter (where it originated), SciPy, and some other large projects. We don't have a meaning for this term in `meson-python` and I think the project is way too small to deal with those kinds of formalities. 

So, I'm fine with using "Funding". 

> I'm not even sure we need the company logo.

I don't really mind either way. It has been a substantial investment, so I'd say showing a logo is justified. But it's not something I asked for so we can leave it out for the time being. 

In case the funding section grows we may need some criteria about when to list something, but let's cross that bridge if and when we get to it.

Comment (dnicolodi) [reply docs/about-us.rst:36]: I'm not at all opposed to having the logo. It was not clear to me the reason why the affiliation of the maintainers is reported. If it is to give back a bit in term of visibility I think having the logo makes sense.

Comment (rgommers) [reply docs/about-us.rst:36]: Great, thanks. In that case I'd prefer keeping it.

File: docs/about-us.rst
Comment (FFY00) [reply docs/about-us.rst:13]: SPDX defines a canonical text for each identifier, that's why it isn't mentioned there. They do define several other variants of the MIT license.

From https://www.gnu.org/licenses/license-list.html#Expat
> Some people call this license “the MIT License,” but that term is misleading, since MIT has used many licenses for software. It is also ambiguous, since the same people also call the [X11 license](https://www.gnu.org/licenses/license-list.html#X11License) “the MIT License,” failing to distinguish them. We recommend not using the term “MIT License.”

But anyway, I really don't care much, so I will remove it.

File: docs/about-us.rst
Comment (FFY00) [reply docs/about-us.rst:53]: Sorry, I meant that IMO it looks signifiantly better *with* it, not *without*.

**Comment by rgommers** [comment]
This is superseded by gh-445, so I will close this PR. Thank you for the work on this @FFY00 and the review @dnicolodi.

**Pull request closed without merging**
